File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;

        for(int i = 0; i < size(nums); i++) {
            int complement = target - nums[i];
            if(hash.find(complement) != hash.end()) {
                return {hash[complement],i};
            }
            hash[nums[i]] = i; 
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        freq ={}

        for i, num in enumerate(nums):
            c = target - num
            if c in freq:
                return [freq[c], i]
            freq[num] = i
        return []
